File: examples/15_sympy/main.cpp
#include <codac>
#include <codac-sympy.h>

using namespace std;
using namespace codac2;

int main()
{
  ScalarVar x,y;
  AnalyticFunction f({x,y}, x*y + sin(x));

  auto dfdx = sympy_partial_diff(f,x);
  auto dfdy = sympy_partial_diff(f,y);

  cout << dfdx << endl;
  cout << dfdy << endl;
}

File: examples/15_sympy/main.py
from codac import *

x = ScalarVar()
y = ScalarVar()
f = AnalyticFunction([x,y], x*y + sin(x))

dfdx = sympy_partial_diff(f, x)
dfdy = sympy_partial_diff(f, y)

print(dfdx)
print(dfdy)